lib/internal/http2.js

```diff
@@ -1629,6 +1611,22 @@ class Http2ClientRequest extends Http2Outgoing {
   }
 }
 
+function addTrailers () {
+  const request = this[kRequest];
+  if (request[kTrailers] instanceof Map) {
```

Comment:
should we use the internal util for this? Is there a chance that the Map was created in a different context?